src/components/VideoPlayer/index.js

```diff
@@ -0,0 +1,167 @@
+/* eslint-disable no-underscore-dangle */
+import {ResizeMode, Video} from 'expo-av';
+import PropTypes from 'prop-types';
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+import {View} from 'react-native';
+import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import {usePlaybackContext} from '@components/VideoPlayerContexts/PlaybackContext';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import addEncryptedAuthTokenToURL from '@libs/addEncryptedAuthTokenToURL';
+import styles from '@styles/styles';
+import VideoPlayerControls from './VideoPlayerControls';
+
+const propTypes = {
+    url: PropTypes.string.isRequired,
+
+    shouldPlay: PropTypes.bool,
+
+    onVideoLoaded: PropTypes.func,
+
+    resizeMode: PropTypes.string,
+
+    isLooping: PropTypes.bool,
+
+    // eslint-disable-next-line react/forbid-prop-types
+    style: PropTypes.arrayOf(PropTypes.object),
```

Comment:
Not sure if this will work here, but we have a special props for that:
```javascript
import stylePropTypes from '@styles/stylePropTypes';
```
```suggestion
    style: stylePropTypes,
```